test/Microsoft.SqlTools.ServiceLayer.Test/LanguageServer/PeekDefinitionTests.cs

```diff
@@ -0,0 +1,228 @@
+
+            // set up workspace mock
+            workspaceService = new Mock<WorkspaceService<SqlToolsSettings>>();
+            workspaceService.Setup(service => service.Workspace.GetFile(It.IsAny<string>()))
+                .Returns(fileMock.Object);
+
+            // setup binding queue mock
+            bindingQueue = new Mock<ConnectedBindingQueue>();
+            bindingQueue.Setup(q => q.AddConnectionContext(It.IsAny<ConnectionInfo>()))
+                .Returns(this.testConnectionKey);
+
+            // inject mock instances into the Language Service
+            LanguageService.WorkspaceServiceInstance = workspaceService.Object;
+            LanguageService.ConnectionServiceInstance = TestObjects.GetTestConnectionService();
+            ConnectionInfo connectionInfo = TestObjects.GetTestConnectionInfo();
+            LanguageService.ConnectionServiceInstance.OwnerToConnectionMap.Add(this.testScriptUri, connectionInfo);
+            LanguageService.Instance.BindingQueue = bindingQueue.Object;
+
+            // setup the mock for SendResult
+            requestContext = new Mock<RequestContext<Location[]>>();
+            requestContext.Setup(rc => rc.SendResult(It.IsAny<Location[]>()))
+                .Returns(Task.FromResult(0));
+
+            // setup the IBinder mock
+            binder = new Mock<IBinder>();
+            binder.Setup(b => b.Bind(
+                It.IsAny<IEnumerable<ParseResult>>(),
+                It.IsAny<string>(),
+                It.IsAny<BindMode>()));
+
+            var testScriptParseInfo = new ScriptParseInfo();
+            LanguageService.Instance.AddOrUpdateScriptParseInfo(this.testScriptUri, testScriptParseInfo);
+            testScriptParseInfo.IsConnected = true;
+            testScriptParseInfo.ConnectionKey = LanguageService.Instance.BindingQueue.AddConnectionContext(connectionInfo);
+
+            // setup the binding context object
+            ConnectedBindingContext bindingContext = new ConnectedBindingContext();
+            bindingContext.Binder = binder.Object;
+            bindingContext.MetadataDisplayInfoProvider = new MetadataDisplayInfoProvider();
+            LanguageService.Instance.BindingQueue.BindingContextMap.Add(testScriptParseInfo.ConnectionKey, bindingContext);
+        }
+
+
+        /// <summary>
+        /// Tests the definition event handler. When called with no active connection, no definition is sent
+        /// </summary>
+        [Fact]
+        public void DefinitionsHandlerWithNoConnectionTest()
+        {
+
+            InitializeTestObjects();
+
+            // request the completion list
+            Task handleCompletion = LanguageService.HandleDefinitionRequest(textDocument, requestContext.Object);
+            handleCompletion.Wait(TaskTimeout);
+
+            // verify that send result was not called
+            requestContext.Verify(m => m.SendResult(It.IsAny<Location[]>()), Times.Never());
+
+        }
+
+#if LIVE_CONNECTION_TESTS
```

Comment:
Yes, these need real SMO to test the scripting and the declarations. These get run when you build and run tests with the integration option. 
Yeah we should be making these required, there are a bunch of tests that are marked live connection.